Library/Homebrew/software_spec.rb

```diff
@@ -50,6 +50,11 @@ def initialize(flags: [])
     @uses_from_macos_elements = []
   end
 
+  def initialize_copy(other)
```

Comment:
@Bo98 bizarrely, I've narrowed this addition down to causing the `installed_dependents_spec` test failures (even on macOS). Mind taking a look?

Reply:
From an initial glance, it looks caused by `Formula["bar"].class.depends_on` hacks that test contains, which after this change will not affect existing instances of `Formula["bar"]` - only new instances.

The changes I made here strongly assume `depends_on` is not called after a formula is loaded for the first time - otherwise you'll unravel a whole load of mess to deal with such as the Formulary cache.

I still feel it's reasonable to not support that scenario, so the test should probably be refactored to instead define different formula stubs with the necessary `depends_on` lines present right from the start.

Reply:
> I still feel it's reasonable to not support that scenario, so the test should probably be refactored to instead define different formula stubs with the necessary `depends_on` lines present right from the start.

Yes, sounds good to me, definitely would rather the test is fixed rather than the code changed. @Bo98 if you can (force?) push that here: it'd be great.

> The changes I made here strongly assume `depends_on` is not called after a formula is loaded for the first time - otherwise you'll unravel a whole load of mess to deal with such as the Formulary cache.

For a follow-up PR: this should probably raise an exception.

Reply:
Will take a look shortly

Reply:
Fixed hopefully: https://github.com/Homebrew/brew/compare/bac61775f73579608c97c8487b0de93620af0298..7c6f7dffa567b3bd91d93c6e22b5bc58648f6a01